src/keeshare/ShareExport.cpp

```diff
@@ -108,124 +106,106 @@ namespace
-                    ShareExport::tr("Could not write export container (%1)").arg(zip.getZipError())};
-        }
-        {
-            QuaZipFile file(&zip);
-            const auto signatureOpened = file.open(QIODevice::WriteOnly, QuaZipNewInfo(KeeShare::signatureFileName()));
-            if (!signatureOpened) {
-                ::qWarning("Embedding signature failed: Could not open file to write (%d)", zip.getZipError());
-                return {reference.path,
-                        ShareObserver::Result::Error,
-                        ShareExport::tr("Could not embed signature: Could not open file to write (%1)")
-                            .arg(file.getZipError())};
-            }
-            QTextStream stream(&file);
-            KeeShareSettings::Sign sign;
-            // TODO: check for false return
-            Signature::create(bytes, own.key.key, sign.signature);
-            sign.certificate = own.certificate;
-            stream << KeeShareSettings::Sign::serialize(sign);
-            stream.flush();
-            if (file.getZipError() != ZIP_OK) {
-                ::qWarning("Embedding signature failed: Could not write file (%d)", zip.getZipError());
-                return {
-                    reference.path,
-                    ShareObserver::Result::Error,
-                    ShareExport::tr("Could not embed signature: Could not write file (%1)").arg(file.getZipError())};
-            }
-            file.close();
-        }
-        {
-            QuaZipFile file(&zip);
-            const auto dbOpened = file.open(QIODevice::WriteOnly, QuaZipNewInfo(KeeShare::containerFileName()));
-            if (!dbOpened) {
-                ::qWarning("Embedding database failed: Could not open file to write (%d)", zip.getZipError());
-                return {reference.path,
-                        ShareObserver::Result::Error,
-                        ShareExport::tr("Could not embed database: Could not open file to write (%1)")
-                            .arg(file.getZipError())};
-            }
-            file.write(bytes);
-            if (file.getZipError() != ZIP_OK) {
-                ::qWarning("Embedding database failed: Could not write file (%d)", zip.getZipError());
-                return {reference.path,
-                        ShareObserver::Result::Error,
-                        ShareExport::tr("Could not embed database: Could not write file (%1)").arg(file.getZipError())};
-            }
-            file.close();
-        }
-        zip.close();
-        return {reference.path};
-#endif
+        zipOpenNewFileInZip64(zf,
+                              fileName.toLatin1().data(),
+                              nullptr,
+                              nullptr,
+                              0,
+                              nullptr,
+                              0,
+                              nullptr,
+                              Z_DEFLATED,
+                              Z_DEFAULT_COMPRESSION,

```

Comment:
This should be `Z_BEST_COMPRESSION`. There is absolutely no reason to use any compression level other than 9 with GZip/Deflate.

Reply:
Technically it doesn't matter because you are compressing random data.